workers/unity/Packages/io.improbable.gdk.gameobjectcreation/EntityTypeExpectations.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Collections.Generic;
+
+namespace Improbable.Gdk.GameObjectCreation
+{
+    /// <summary>
+    ///     Holds mappings from entity types to the set of components expected before creating their GameObjects.
+    /// </summary>
+    public class EntityTypeExpectations
+    {
+        private Type[] defaultExpectation;
+
+        private readonly Dictionary<string, Type[]> entityExpectations
+            = new Dictionary<string, Type[]>();
+
+        public void RegisterDefault(Type[] defaultComponentTypes)
+        {
+            defaultExpectation = defaultComponentTypes;
+        }
+
+        public void RegisterEntityType(string entityType, Type[] expectedComponentTypes)
+        {
+            entityExpectations.Add(entityType, expectedComponentTypes ?? new Type[] { });
+        }
```

Comment:
If the second parameter is optional, make it default `null`. If its not optional, it should be an exception